### PR TITLE
Refine README

### PR DESCRIPTION
**Pull request by faebr**
Wanted to fix a typo but then ended up improving the section on running it locally. Added different sections for running the operator on a local kind cluster or on the local machine with only NetBox running on kind.

Changes added:
- Streamline the naming to "NetBox Operator" for the speaking name and "netbox-operator" for URI/filesystem compatible name.
- Match diagrams with sample yamls and move to GitHub compatible drawio svg file
- Add notes on how to run on podman
- Add example on how to run with demo.netbox.dev

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -17,33 +17,49 @@ NetBox Operator extends the Kubernetes API by allowing users to manage NetBox re
...
-    NETBOX_RESTORATION_HASH_FIELD_NAME="netboxOperatorRestorationHash"
+    export NETBOX_HOST="localhost:8080"
+    export AUTH_TOKEN="0123456789abcdef0123456789abcdef01234567"
+    export POD_NAMESPACE="default"
+    export HTTPS_ENABLE="false"
+    export NETBOX_RESTORATION_HASH_FIELD_NAME="netboxOperatorRestorationHash"
     ```
-- execute `make deploy-kind`
-- in a separate terminal, run `kubectl port-forward deploy/netbox 8080:8080 -n default`
+- run `make run` in a new terminal to start the netbox operator locally and accept incoming connections if a popup apears
```
Comment (henrybear327) [inline README.md:44]: Typo: appears

**Review by henrybear327 — CHANGES_REQUESTED** [review]

File: README.md [hunk 2]
```diff
@@ -17,33 +17,49 @@ NetBox Operator extends the Kubernetes API by allowing users to manage NetBox re
...
+### Running the Operator locally sing an existing NetBox instance and Kubernetes cluster
 
 Prerequisites:
 - a NetBox instance to test against
 - a Kubernetes cluster with the netbox-operator CRDs installed (point the kubeconfig to the cluster and run `make install`)
-
-There are some mandatory environment variables to set to run the netbox-operator locally: `NETBOX_HOST`, `AUTH_TOKEN` and `POD_NAMESPACE`.
-
-If you want to run the operator against a host with HTTPS protocol you need to set the `HTTPS_ENABLE=true`.
+- There are some mandatory environment variables to set to run the netbox-operator locally. Make sure they are adapted to your NetBox instance and your NetBox instance is reachable using the defined HOST:
```
Comment (henrybear327) [inline README.md:54]: Maybe we would like to unify the usages of operator, netbox operator, and netbox-operator in this PR :)

**Review by henrybear327 — CHANGES_REQUESTED** [review]

File: README.md (hunk 1 above)
Comment (faebr) [reply README.md]: thanks for spotting, fixed in https://github.com/netbox-community/netbox-operator/pull/41/commits/22d63ccdbbdfd9e5aba614f6b920dc4d03507461

File: README.md (hunk 2 above)
Comment (faebr) [reply README.md]: usage unified to NetBox-Operator in https://github.com/netbox-community/netbox-operator/pull/41/commits/22d63ccdbbdfd9e5aba614f6b920dc4d03507461

File: README.md [hunk 3]
```diff
@@ -17,33 +17,49 @@ NetBox Operator extends the Kubernetes API by allowing users to manage NetBox re
...
+- a Kubernetes cluster with the NetBox-Operator CRDs installed (point the kubeconfig to the cluster and run `make install`)
+- There are some mandatory environment variables to set to run the NetBox-Operator locally. Make sure they are adapted to your NetBox instance and your NetBox instance is reachable using the defined HOST:
+    ```
+    export NETBOX_HOST="localhost:8080"
+    export AUTH_TOKEN="0123456789abcdef0123456789abcdef01234567"
+    export POD_NAMESPACE="default"
+    export HTTPS_ENABLE="true"
+    export NETBOX_RESTORATION_HASH_FIELD_NAME="netboxOperatorRestorationHash"
+    ```
+- run `make run` in a new terminal to start the NetBox-Operator locally and accept incoming connections if a popup apears
```
Comment (henrybear327) [inline README.md:62]: `typo`: apears :D

Comment (faebr) [reply README.md]: -.- ... fixed in https://github.com/netbox-community/netbox-operator/pull/41/commits/d39768e96ab2a62e9e2be50d5e1d22245bf6d6f0

File: README.md [hunk 4]
```diff
@@ -64,13 +98,14 @@ make install
 ### Deploy the Manager to the cluster with the image specified by `IMG`
 
 ```sh
-make deploy IMG=<some-registry>/netbox-operator:tag
+make deploy IMG=<some-registry>/NetBox Operator:tag
```
Comment (alexandernorth) [inline README.md:101]: This should probably be left as is

File: README.md [hunk 5]
```diff
@@ -12,45 +12,79 @@ NetBox Operator extends the Kubernetes API by allowing users to manage NetBox re
...
+    export POD_NAMESPACE="default"
+    export HTTPS_ENABLE="true"
+    export NETBOX_RESTORATION_HASH_FIELD_NAME="netboxOperatorRestorationHash"
     ```
-- execute `make deploy-kind`
-- in a separate terminal, run `kubectl port-forward deploy/netbox 8080:8080 -n default`
-- go to your favorite browser and type in `localhost:8080`, with the username `admin` and password `admin`, you will be able to access the local NetBox instance running in the kind cluster
 
-### Using an existing NetBox instance and Kubernetes cluster
+- Run the NetBox Operator locally (in a new terminal window) `make install && make run`
```
Comment (alexandernorth) [inline README.md:67]: This should be in the same terminal as the export statements are run else they'll not be picked up

File: README.md [hunk 6]
```diff
@@ -12,45 +12,79 @@ NetBox Operator extends the Kubernetes API by allowing users to manage NetBox re
...
+    export HTTPS_ENABLE="false"
+    export NETBOX_RESTORATION_HASH_FIELD_NAME="netboxOperatorRestorationHash"
     ```
-    NETBOX_HOST="localhost:8080"
-    AUTH_TOKEN="0123456789abcdef0123456789abcdef01234567"
-    POD_NAMESPACE="default"
-    HTTPS_ENABLE="false"
-    NETBOX_RESTORATION_HASH_FIELD_NAME="netboxOperatorRestorationHash"
+
+- Run the NetBox Operator locally (in a new terminal window) `make install && make run`
```
Comment (alexandernorth) [inline README.md:47]: This should be in the same terminal as the export statements are run else they'll not be picked up

File: README.md [hunk 7]
```diff
@@ -116,7 +153,7 @@ Following are the steps to build the installer and distribute this project to us
 1. Build the installer for the image built and published in the registry:
 
 ```sh
-make build-installer IMG=<some-registry>/netbox-operator:tag
+make build-installer IMG=<some-registry>/NetBox Operator:tag
```
Comment (alexandernorth) [inline README.md:156]: this should be left as is

File: README.md [hunk 8]
```diff
@@ -129,11 +166,12 @@ its dependencies.
 Users can just run `kubectl apply -f <URL for YAML BUNDLE>` to install the project, i.e.:
 
 ```sh
-kubectl apply -f https://raw.githubusercontent.com/<org>/netbox-operator/<tag or branch>/dist/install.yaml
+kubectl apply -f https://raw.githubusercontent.com/<org>/NetBox Operator/<tag or branch>/dist/install.yaml
```
Comment (alexandernorth) [inline README.md:169]: this should be left as is

File: README.md [hunk 9]
```diff
@@ -12,45 +12,79 @@ NetBox Operator extends the Kubernetes API by allowing users to manage NetBox re
...
+## Running and developing locally
+
+### Running the NetBox Operator and NetBox on a local kind cluster
+
+Note: This requires Docker BuildKit.
+
+- Create kind cluster with a NetBox deployment: `make create-kind`
+- Port-forward NetBox: `kubectl port-forward deploy/netbox 8080:8080 -n default`
+- Open <http://localhost:8080> in your favorite browser and log in with the username `admin` and password `admin`, you will be able to access the local NetBox instance running in the kind cluster.
+- Deploy the NetBox Operator on the local kind cluster: `make deploy-kind`
```
Comment (alexandernorth) [inline README.md:30]:
```suggestion
- Create kind cluster with a NetBox deployment: `make create-kind`
- Deploy the NetBox Operator on the local kind cluster: `make deploy-kind`

To optionally access the NetBox UI:
- Port-forward NetBox: `kubectl port-forward deploy/netbox 8080:8080 -n default`
- Open <http://localhost:8080> in your favorite browser and log in with the username `admin` and password `admin`, you will be able to access the local NetBox instance running in the kind cluster.
```

Reorder steps and added separation for accessing the UI

**Review by alexandernorth — CHANGES_REQUESTED** [review]

File: README.md [hunk 10]
```diff
@@ -1,49 +1,87 @@
...
 
 **Disclaimer:** This project is currently under development and may change rapidly, including breaking changes. Use with caution in production environments.
 
 NetBox Operator extends the Kubernetes API by allowing users to manage NetBox resources – such as IP addresses and prefixes – directly through Kubernetes. This integration brings Kubernetes-native features like reconciliation, ensuring that network configurations are maintained automatically, thereby improving both efficiency and reliability.
 
-<div align=center>
-    <img src="./docs/NetBox Operator High-Level Architecture.png" alt="Diagram: NetBox Operator High-Level Architecture" width="800"/>
-    <p><em>Figure 1: NetBox Operator High-Level Architecture</em></p>
-</div>
+![Figure 1: NetBox Operator High-Level Architecture](docs/netbox-operator-high-level-architecture.drawio.svg)
```
Comment (alexandernorth) [inline README.md:7]: This looks odd with github in dark mode - maybe you could add a white background? also I think that there is a missing icon for k8s

image

File: README.md [hunk 11]
```diff
@@ -1,49 +1,87 @@
...
+
 - go version v1.22.0+
-- docker version 17.03+.
-- kubectl version v1.11.3+.
-- Access to a Kubernetes v1.11.3+ cluster.
+- docker version 17.03+
+- kubectl version v1.11.3+
+- Access to a Kubernetes v1.11.3+ cluster
+- kind
+- docker cli
```
Comment (alexandernorth) [inline README.md:18]: Is this not included with docker by default? Does no harm to keep though

File: README.md [hunk 12]
```diff
@@ -1,49 +1,87 @@
...
+## Testing NetBox Operator using samples
+
+In the folder `config/samples/` you can find example manifests to create IpAddress, IpAddressClaim, Prefix and PrefixClaim resources. Apply them to the cluster with `kubectl apply -f <file-name>` and use your favorite Kubernetes tools to displa.
+
+Example of assigning a Prefix using PrefixClaim:
 
-Prerequisites:
-- a NetBox instance to test against
-- a Kubernetes cluster with the netbox-operator CRDs installed (point the kubeconfig to the cluster and run `make install`)
+![Figure 2: PrefixClaim example with a NetBox and NetBox Operator instance deployed on the same cluster](docs/prefixclaim-sample-with-netbox-running-in-cluster.drawio.svg)
```
Comment (alexandernorth) [inline README.md:77]: Same issues as previous figure

File: README.md [hunk 13]
```diff
@@ -1,49 +1,87 @@
...
 
-There are some mandatory environment variables to set to run the netbox-operator locally: `NETBOX_HOST`, `AUTH_TOKEN` and `POD_NAMESPACE`.
+1. Apply a PrefixClaim: `kubectl apply -f config/samples/netbox_v1_prefixclaim.yaml`
+2. Wait for ready condition: `kubectl wait prefix prefixclaim-sample --for=condition=Ready`
+3. List PrefixClaim and Prefix resources: `kubectl get pxc,px`
+4. For the prefix you’ll be able to see the URL, you can open it (login is admin/admin) to browse through the NetBox resources.
```
Comment (alexandernorth) [inline README.md:82]:
```suggestion
4. In the prefix status fields you’ll be able to see the netbox URL of the resource. Login with the default `admin`/`admin` credentials to access the NetBox resources.
```

File: README.md [hunk 14]
```diff
@@ -1,49 +1,87 @@
...
 
-If you want to run the operator against a host with HTTPS protocol you need to set the `HTTPS_ENABLE=true`.
+Make sure to also discover the yaml output of these resources, check the events and Operator logs.
```
Comment (alexandernorth) [inline README.md:84]:
```suggestion
Key information can be found in the yaml formatted output of these resources, as well as in the events and Operator logs.
```

File: README.md [hunk 15]
```diff
@@ -101,11 +140,13 @@ make undeploy
 
 ## Restoration from NetBox
 
-In case the cluster that contains the NetBox Custom Resources managed by this Operator is not backed up (e.g. using Velero), we need to be able to restore some information from NetBox. This includes two mechanisms implemented in this Operator:
-- `IpAddressClaim` and `PrefixClaim` have the flag `preserveInNetbox` in their spec. If set to true, the Operator will not delete the assigned IP Address/Prefix in NetBox when the Kubernetes Custom Resource is deleted
+In case the cluster that contains the NetBox Custom Resources managed by this NetBox Operator is not backed up (e.g. using Velero), we need to be able to restore some information from NetBox. This includes two mechanisms implemented in this NetBox Operator:
```
Comment (alexandernorth) [inline README.md:143]:
```suggestion
In the case that the cluster containing the NetBox Custom Resources managed by this NetBox Operator is not backed up (e.g. using Velero), we need to be able to restore some information from NetBox. This includes two mechanisms implemented in this NetBox Operator:
```

**Review by alexandernorth — CHANGES_REQUESTED** [review]

File: README.md (hunk 11 above)
Comment (jstudler) [reply README.md:18]: Let's keep it since there may be users that want to use another runtime such as podman.

**Review by alexandernorth — APPROVED** [review]
lgtm!

File: README.md [hunk 16]
```diff
@@ -1,49 +1,87 @@
...
-- execute `make deploy-kind`
-- in a separate terminal, run `kubectl port-forward deploy/netbox 8080:8080 -n default`
-- go to your favorite browser and type in `localhost:8080`, with the username `admin` and password `admin`, you will be able to access the local NetBox instance running in the kind cluster
 
-### Using an existing NetBox instance and Kubernetes cluster
+- Run the NetBox Operator locally `make install && make run`
+
+## Testing NetBox Operator using samples
+
+In the folder `config/samples/` you can find example manifests to create IpAddress, IpAddressClaim, Prefix and PrefixClaim resources. Apply them to the cluster with `kubectl apply -f <file-name>` and use your favorite Kubernetes tools to displa.
```
Comment (henrybear327) [inline README.md:73]: Typo `displa`

**Review by henrybear327 — CHANGES_REQUESTED** [review]

**Comment by henrybear327** [comment]
LGTM! Just a minor typo to be fixed before merging :)

**Review by henrybear327 — APPROVED** [review]

File: README.md
Comment (bruelea) [inline README.md:1]: We could remove this line, `>**NOTE**: Ensure that the samples have default values to test it out.` as the samples in the project have default values set.

**Review by bruelea — APPROVED** [review]

**Pull request merged**